pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,232 @@
+package nodelabeller
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+const (
+	labelNamespace    = "feature.node.kubernetes.io"
+	labellerNamespace = "node-labeller"
+)
+
+var nodeLabellerVolumePath = "/var/lib/kubevirt-node-labeller"
+
+//NodeLabeller struct holds informations needed to run node-labeller
+type NodeLabeller struct {
+	kvmController     *device_manager.DeviceController
+	configMapInformer cache.SharedIndexInformer
+	nodeInformer      cache.SharedIndexInformer
+	clientset         kubecli.KubevirtClient
+	Queue             workqueue.RateLimitingInterface
+	host              string
+	namespace         string
+	logger            *log.FilteredLogger
+}
+
+func NewNodeLabeller(kvmController *device_manager.DeviceController, nodeInformer, configMapInformer cache.SharedIndexInformer, clientset kubecli.KubevirtClient, host, namespace string) *NodeLabeller {
+	return &NodeLabeller{
+		kvmController:     kvmController,
+		configMapInformer: configMapInformer,
+		nodeInformer:      nodeInformer,
+		clientset:         clientset,
+		Queue:             workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+
+		host:      host,
+		namespace: namespace,
+		logger:    log.DefaultLogger(),
+	}
+}
+
+//Run runs node-labeller
+func (n *NodeLabeller) Run(threadiness int, stop chan struct{}) {
+	defer n.Queue.ShutDown()
+	n.logger.Infof("node-labeller is running")
+
+	if !n.kvmController.NodeHasDevice(device_manager.KVMPath) {
+		n.logger.Infof("Node-labeller cannot work without KVM device.")
+		return
+	}
+
+	n.configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(_ interface{}) {
+			n.Queue.Add(n.host)
+		},
+		DeleteFunc: func(_ interface{}) {
+			n.Queue.Add(n.host)
+		},
+		UpdateFunc: func(_, _ interface{}) {
+			n.Queue.Add(n.host)
+		},
+	})
+
+	go n.configMapInformer.Run(stop)
+
+	n.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(_, _ interface{}) {
+			n.Queue.Add(n.host)
```

Comment:
Is this informer using a field selector which only lists the node where we are running on?

Reply:
Yes https://github.com/kubevirt/kubevirt/pull/3133/files#diff-c2036087b983cb9d7ed6695ae5464c83R189

Reply:
:+1: